config/clusters/2i2c/binderhub-ui-demo.values.yaml

```diff
@@ -0,0 +1,114 @@
+jupyterhub:
+  ingress:
+    hosts:
+      - hub.binderhub-ui-demo.2i2c.cloud
+    tls:
+      - secretName: https-auto-tls
+        hosts:
+          - hub.binderhub-ui-demo.2i2c.cloud
+  custom:
+    2i2c:
+      add_staff_user_ids_to_admin_users: true
+      add_staff_user_ids_of_type: "google"
+    jupyterhubConfigurator:
+      enabled: false
+    binderhubUI:
+      enabled: true
+    homepage:
+      templateVars:
+        org:
+          name: Demo binderhub UI with binderhub-service
+          url: https://2i2c.org
+          logo_url: https://2i2c.org/media/logo.png
+        designed_by:
+          name: 2i2c
+          url: https://2i2c.org
+        operated_by:
+          name: 2i2c
+          url: https://2i2c.org
+        funded_by:
+          name: ""
+          url: ""
+    singleuserAdmin:
+      extraVolumeMounts: []
+  singleuser:
+    storage:
+      type: none
+      extraVolumeMounts: []
+    initContainers: []
+  hub:
+    redirectToServer: false
+    services:
+      binder:
+        oauth_no_confirm: true
+        oauth_redirect_uri: https://binderhub-ui-demo.2i2c.cloud/oauth_callback
+    loadRoles:
+      binder:
+        services:
+          - binder
+        scopes:
+          - servers
+          - read:users # admin:users is required if authentication isn't enabled
+      user:
+        scopes:
+          - self
+          # Admin users will by default have access:services, so this is only
+          # observed to be required for non-admin users.
+          - access:services!service=binder
+    config:
+      BinderSpawnerMixin:
+        auth_enabled: true
+      JupyterHub:
+        authenticator_class: cilogon
+      CILogonOAuthenticator:
+        oauth_callback_url: "https://hub.binderhub-ui-demo.2i2c.cloud/hub/oauth_callback"
+        allowed_idps:
+          http://google.com/accounts/o8/id:
+            username_derivation:
+              username_claim: "email"
+binderhub-service:
+  enabled: true
+  ingress:
+    enabled: true
+    hosts:
+      - binderhub-ui-demo.2i2c.cloud
+    tls:
+      - secretName: binder-https-auto-tls
+        hosts:
+          - binderhub-ui-demo.2i2c.cloud
+  config:
+    BinderHub:
+      base_url: /
+      hub_url: https://hub.binderhub-ui-demo.2i2c.cloud
+      badge_base_url: https://binderhub-ui-demo.2i2c.cloud
+      auth_enabled: true
+      enable_api_only_mode: false
+      image_prefix: us-central1-docker.pkg.dev/two-eye-two-see/binderhub-ui-demo-registry/binderhub-service-
+  extraConfig:
+    # FIXME: set KubernetesBuildExecutor.push_secret again
+    #        without this for some reason the build pods
+    #        search after the wrong secret name (i.e. the default name)
+    #        set by binderhub in KubernetesBuildExecutor.push_secret
+    01-binderhub-service-set-push-secret: |
+      import os
+      c.KubernetesBuildExecutor.push_secret = os.environ["PUSH_SECRET_NAME"]
```

Comment:
Every time I work with binderhub, the `push_secret` config is a big confusion - here are reasons for why its confusing observed in the last ~5 minutes looking:

1. `push_secret` should be configured to a k8s secret name having a `config.json` key providing a docker configuration including credentials to work against a registry (its not the typical k8s imagePullSecret or similar, its a docker client config)
2. `binder-build-docker-config` is a default value for `push_secret`, but its also hardcoded i the official BinderHub helm chart deployment, but in binderhub-service deployment it isn't hardcoded - instead `push_secret` is set by a `extraConfig` entry based on an env variable etc - we shouldn't touch `push_secret` config as users of binderhub-service I think.
3. In binderhub-service chart, we configure credentials that goes into the k8s Secret via chart config `buildPodsRegistryCredentials`, but in the official chart its under config.BinderHub.buildDockerConfig` which is confusing because it isn't a traitlet for the Python class `BinderHub` as expected if listed there.
4. The Deployment resource isn't changed if the credentials under buildDockerConfig

Reply:
TL;DR - I was confused, and remain confused why you observed that.